packages/xrpl/src/sugar/submit.ts

```diff
@@ -222,10 +223,26 @@ async function waitForFinalTransactionOutcome(
 // checks if the transaction has been signed
 function isSigned(transaction: Transaction | string): boolean {
   const tx = typeof transaction === 'string' ? decode(transaction) : transaction
-  return (
-    typeof tx !== 'string' &&
-    (tx.SigningPubKey != null || tx.TxnSignature != null)
-  )
+  if (typeof tx === 'string') {
+    return false
+  }
+  if (tx.Signers != null) {
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- we know that tx.Signers is an array of Signers
+    const signers = tx.Signers as Signer[]
+    for (const signer of signers) {
```

Comment:
This will only check the first signer and not all of them

Reply:
The `for of` loop checks each signer by iterating through all signers.